### PR TITLE
cd $HOME when no arguments are passed.

### LINKED ISSUE #7: `cd` without arguments should take to $HOME
Currently it blows up.

### PR DESCRIPTION
**Pull request by ordovicia**
This PR fixes #7.

### CODE REVIEW

File: src/shell/builtins/cd.rs [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
 pub fn run(args: Vec<String>) -> Result<()> {
-    let target = Path::new(&args[0]);
-
-    env::set_current_dir(&target)?;
+    if args.is_empty() {
+        let target: PathBuf = env::home_dir().expect("Could not get your home dir");
+        env::set_current_dir(&target)?;
+    } else {
+        let target: &Path = Path::new(&args[0]);
+        env::set_current_dir(target)?;
```
Comment (juanibiapina) [inline src/shell/builtins/cd.rs]: `set_current_dir` appears on both clauses of the `if`, so you can move it out and prevent the duplication. Let the `if` statement only take care of selecting the target directory.

**Comment by juanibiapina** [comment]
The code looks great. Before I approve, I would ask you to reword the commit message for the second commit. You never need to say 'Modify' in the commit message, because you usually modify anyway. The import verb there would be 'Remove'.

Or you can just squash it into the first commit and have only one, your choice.

Have fun at hacktoberfest!

**Comment by ordovicia** [comment]
Thank you for approving and advice about commit message!

I would like to squash commits.
Could I `push -f`?


Originally, I thought there was a reason to duplicate `set_current_dir()`:
`env::home_dir()` returns `PathBuf`.
On the other hand, `Path::new()` returns `Path`, which is a reference of `Path`.
So values of type `Path` should be converted into a value of type `PathBuf` by `to_path_buf()`.
I thought this conversion is ineffective.

**Comment by juanibiapina** [comment]
Yes, `push -f`

**Comment by juanibiapina** [comment]
You're right, the code is slightly less efficient, but it reads better. My first concern is usually good readability, then performance.

Now that we have the code reading well, we can think on how to do that with less allocations without getting on the way of readability.

If you feel like it is worth the investigation, you can do the optimization in this PR (or even a new one if you think you've used enough time investigating.)

**Comment by ordovicia** [comment]
Squashed and pushed.

Thank you for your comment!

**Pull request merged**